app/server/sonicpi/lib/sonicpi/lang/core.rb

```diff
@@ -2511,11 +2511,23 @@ def wait(time)
 
 
 
+      def immutable?(v)
+        return true if v.is_a?(Numeric) || v.is_a?(Symbol) || v.is_a?(TrueClass) || v.is_a?(FalseClass) || v.is_a?(NilClass)
+        if v.is_a?(SonicPi::Core::SPVector)
+          v.each do |el|
+            return false if not immutable?(el)
+          end
+          return true
```

Comment:
one way to shorten those 4 lines would be something like `return v.all? { |el| immutable?(el) }`